feat: реализовать хранение данных в реляционной БД

Alex-Galvinauskas opened the pull request:
- Добавлены зависимости H2 Database и Spring JDBC
- Настроена схема БД через schema.sql с индексами
- Реализованы DAO-классы UserDbStorage и FilmDbStorage с RowMapper
- Обновлены модели данных для соответствия структуре БД
- Реализована новая логика дружбы с системой заявок
- Добавлены эндпоинты для работы с жанрами и MPA рейтингами
- Написаны интеграционные тесты для всех DAO-методов
- Настроено внедрение зависимостей через @Repository и @Primary

Требуется доработать:
- Обновить README.md с описанием новой структуры БД
- Добавить в README.md примеры запросов к новым эндпоинтам
- Описать в документации изменения в логике дружбы

Alex-Galvinauskas committed:
feat: реализовать хранение данных в реляционной БД

- Добавлены зависимости H2 Database и Spring JDBC
- Настроена схема БД через schema.sql с индексами
- Реализованы DAO-классы UserDbStorage и FilmDbStorage с RowMapper
- Обновлены модели данных для соответствия структуре БД
- Реализована новая логика дружбы с системой заявок
- Добавлены эндпоинты для работы с жанрами и MPA рейтингами
- Написаны интеграционные тесты для всех DAO-методов
- Настроено внедрение зависимостей через @Repository и @Primary

Требуется доработать:
- Обновить README.md с описанием новой структуры БД
- Добавить в README.md примеры запросов к новым эндпоинтам
- Описать в документации изменения в логике дружбы

Alex-Galvinauskas committed:
style: исправить переносы строк кода

maks-avilow commented on src/main/java/ru/yandex/practicum/filmorate/controller/GenreController.java:
```diff
@@ -0,0 +1,50 @@
+
+        List<Genre> genres = genreService.getAllGenres();
+
+        log.info("Возвращено {} жанров", genres.size());
```
На самом деле — логирование в контроллерах не обязательно, и во многих проектах его сознательно избегают или сводят к минимуму
Контроллеры — это граница приложения (presentation layer), а не место для бизнес-логики или детального логирования. Их основная задача:
-принять HTTP-запрос
-провалидировать входные данные
-передать управление сервису
-вернуть ответ

Почему часто не логируют в контроллерах:

Дублирование
Если логирование уже есть в сервисах (где происходит реальная работа), то дублировать его на уровне контроллера избыточно.
Проблемы с безопасностью
Логирование тела запроса (@RequestBody) может случайно записать чувствительные данные (пароли, токены, персональную информацию). Это особенно опасно, если логи попадают в централизованные системы (ELK, Splunk и т.п.).
Шум в логах
Контроллеры вызываются часто. Лог каждой операции (особенно DEBUG) быстро засоряет логи, снижая их полезность.
Существуют более подходящие механизмы:
HTTP-инструментарий: access logs веб-сервера (Nginx, Tomcat) или Spring Boot Actuator уже фиксируют все входящие запросы (метод, путь, статус, время).
AOP / фильтры / интерцепторы: можно централизованно логировать все запросы без дублирования кода в каждом контроллере.
Метрики и трассировка: OpenTelemetry, Micrometer — лучше подходят для мониторинга, чем ручное логирование.

Подитожим: это замечание из категории можно лучше, но я считаю его также критичным

maks-avilow commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,20 @@
+
+# JPA
+spring.jpa.hibernate.ddl-auto=none
+spring.jpa.properties.hibernate.format_sql=true
```
да это хорошая практика, плюсую

maks-avilow reviewed (changes requested):
Александр привет! Есть критические замечания, давай еще доработаем

Alex-Galvinauskas committed:
feat: реализовать полноценную DTO-архитектуру

- Все контроллеры теперь работают исключительно с DTO:
  - FilmController, UserController, GenreController, MpaController
  - Абстрактный контроллер типизирован по DTO
  - Принимаем и возвращаем только DTO в запросах/ответах

- Сервисный слой использует мапперы для преобразования DTO-Entity:
  - FilmServiceImpl, UserServiceImpl, GenreServiceImpl, MpaServiceImpl
  - Все методы сервисов работают с DTO на входе и выходе
  - Бизнес-логика оперирует Entity, но границы четко определены

- Добавлены и используются все необходимые мапперы:
  - FilmMapper, UserMapper, GenreMapper, MpaMapper
  - Настроена интеграция с Spring (componentModel = 'spring')

- Все эндпоинты возвращают ResponseEntity:
  - Явное указание HTTP-статусов
  - Единообразный формат ответов
  - Подготовка для будущего расширения (заголовки, кастомные статусы)

- Сохранена валидация на уровне DTO:
  - Аннотации валидации в DTO-классах
  - Бизнес-валидация в отдельных валидаторах
  - Четкое разделение ответственности между слоями

maks-avilow reviewed (approved):
принято
